gunicorn/datadog_checks/gunicorn/gunicorn.py

```diff
@@ -140,6 +152,35 @@ def _get_master_proc_name(name):
         # web      22985 26.4  6.1 795288 449596 ?       Sl   19:30   2:32 gunicorn: worker [web1]
         return "gunicorn: master [%s]" % name
 
+    def _collect_metadata(self):
+        raw_version = self._get_version()
+        self.log.debug('gunicorn version: %s', raw_version)
+
+        if raw_version:
+            self.set_metadata('version', raw_version)
+
+    def _get_version(self):
+        """ Get version from `gunicorn --version` with this format:
+
+                gunicorn (version 19.9.0)
+
+        """
+        try:
+            pc_out, pc_err, _ = get_subprocess_output([self.gunicorn_bin_path, '--version'], self.log, False)
+        except OSError:
+            self.log.warning("Error collecting gunicorn version.")
+            return None
+
+        match = re.match(self.VERSION_PATTERN, pc_out)
+        if not match:
+            match = re.match(self.VERSION_PATTERN, pc_err)
```

Comment:
Is that expected to have the verison in stderr?

Reply:
Yes, seems that's possible. Not sure why though.

But `datadog/agent` image:
```
docker run datadog/agent bash -c 'pip install gunicorn && gunicorn --version 2> /tmp/err 1> /tmp/out && cat -e /tmp/err'
gunicorn (version 19.9.0)$
```

But `python` image:

```
docker run python bash -c 'pip install gunicorn && gunicorn --version 2> /tmp/err 1> /tmp/out && cat -e /tmp/out'
gunicorn (version 19.9.0)$
```

Reply:
Seems like the python container somehow doesn't have stderr working at all:
`docker run python bash -c '>&2 echo "error" 1>/dev/null'` doesn't return anything.

Anyway according to http://docs.gunicorn.org/en/stable/settings.html#errorlog, gunicorn is outputing to stderr by default since 19.2. That means that to be compatible with everything we need to read from stdout and stderr.

Thanks!